hieradata/clients/aws.ci.jenkins.io.yaml

```diff
@@ -403,33 +403,45 @@ profile::jenkinscontroller::jcasc:
             javaHome: /opt/jdk-8
             labels:
               - ubuntu-22-amd64-maven8
-            spot: false
+              - spot
+            spot: true
             acpServerId: aws-internal
           - description: "Spot Ubuntu 22.04 LTS x86_64 with JDK11 set as default java CLI"
-            maxInstances: 1
+            maxInstances: 50
             instanceType: M6aXlarge # 4 vCPUs / 16 Gb
             os: "ubuntu"
             os_version: "22.04"
             ebsOptimized: true
             architecture: "amd64"
             javaHome: /opt/jdk-11
             labels:
-              - ubuntu-22-amd64-maven11-spot
+              - ubuntu-22-amd64-maven11
+              - spot
             spot: true
             acpServerId: aws-internal
-          - description: "Ubuntu 22.04 LTS x86_64 with JDK11 set as default java CLI"
+          ####
+          # Linux VMs jdk17 (default non-spot version for buildPlugin() / docker-*)
+          - description: "Ubuntu 22.04 LTS x86_64 with JDK17 set as default java CLI"
             maxInstances: 50
             instanceType: M6aXlarge # 4 vCPUs / 16 Gb
             os: "ubuntu"
             os_version: "22.04"
             ebsOptimized: true
             architecture: "amd64"
-            javaHome: /opt/jdk-11
+            javaHome: /opt/jdk-17
             labels:
-              - ubuntu-22-amd64-maven11
+              - ubuntu-22-amd64-maven17
+              # Labels indicating it is the default VM template
+              - ubuntu
+              - linux
+              - java
+              - docker
+              - linux-amd64
             spot: false
             acpServerId: aws-internal
-          - description: "Ubuntu 22.04 LTS x86_64 with JDK17 set as default java CLI"
+          ####
+          # Linux VMs jdk17 (default spot version for buildPlugin() / docker-*)
+          - description: "Spot Ubuntu 22.04 LTS x86_64 with JDK17 set as default java CLI"
```

Comment:
We should invert the logic and define "spot" as the default while "ondemand" should be opt-in